src/main/java/io/confluent/connect/elasticsearch/Validator.java

```diff
@@ -117,6 +121,16 @@ private void validateCredentials() {
     }
   }
 
+  private void validateDataStreamFields() {
+    if (config.dataStreamType() == DataStreamType.NONE ^ config.dataStreamDataset() == null) {
+      String errorMessage = String.format(
+          "'%s' must be set if '%s' is set.", DATA_STREAM_DATASET_CONFIG, DATA_STREAM_TYPE_CONFIG
+      );
+      addErrorMessage(DATA_STREAM_TYPE_CONFIG, errorMessage);
+      addErrorMessage(DATA_STREAM_DATASET_CONFIG, errorMessage);
+    }
```

Comment:
we should also validate that WriteMethod is insert and behavior on nulls isnt delete if datastreams is set

Reply:
Should we still validate if only 1 of the 2 data stream configs is set? The way I have it now validates even if only 1 is set.

Reply:
we don't allow only 1 so we can just validate if both are set